db/seeds/remediation.rb

```diff
@@ -0,0 +1,107 @@
+
+# Create the proper form structure
+# This matches the structure expected by Form526Submission#form_to_json
+# form_json should contain a single key Form526Submission::FORM_0781 (likely 'form526')
+# that contains the nested form structure
+form_structure = {
+  'form0781' => {
+    'form0781' => { 'incidents' => [] },
+    'form0781v2' => { 'incidents' => [] }
+  }
+}
+
+test_cases = []
+
+# Case 1: pre-2019 only v1
+case1 = Marshal.load(Marshal.dump(form_structure))
+case1['form0781']['form0781']['incidents'] = [
+  {
+    'incidentDate' => '2018-01-01',
+    'incidentDescription' => 'Pre-threshold case - only 0781',
+    'unitAssigned' => 'Unit1',
+    'unitAssignedDates' => { 'from' => '2018-01-01', 'to' => '2018-01-02' }
+  }
+]
+test_cases << { id: 1, date: Date.new(2018, 1, 1), form_json: case1 }
+
+# Case 2: post-2019 both
+case2 = Marshal.load(Marshal.dump(form_structure))
+case2['form0781']['form0781']['incidents'] = [
+  {
+    'incidentDate' => '2020-01-01',
+    'incidentDescription' => 'Post-threshold both forms 0781',
+    'unitAssigned' => 'Unit2',
+    'unitAssignedDates' => { 'from' => '2020-01-01', 'to' => '2020-01-02' }
+  }
+]
+case2['form0781']['form0781v2']['incidents'] = [
+  {
+    'incidentDate' => '2020-01-01',
+    'incidentDescription' => 'Post-threshold 0781v2',
+    'unitAssigned' => 'Unit2v2',
+    'unitAssignedDates' => { 'from' => '2020-01-01', 'to' => '2020-01-02' }
+  }
+]
+test_cases << { id: 2, date: Date.new(2020, 1, 1), form_json: case2 }
+
+# Case 3: post-2019 only v2
+case3 = Marshal.load(Marshal.dump(form_structure))
+# v1 blank, v2 has
+case3['form0781']['form0781v2']['incidents'] = [
+  {
+    'incidentDate' => '2021-02-02',
+    'incidentDescription' => 'Only 0781v2 content',
+    'unitAssigned' => 'Unit3v2',
+    'unitAssignedDates' => { 'from' => '2021-02-02', 'to' => '2021-02-03' }
+  }
+]
+test_cases << { id: 3, date: Date.new(2021, 2, 2), form_json: case3 }
+
+# Cases 4–6: blank (skip)
+(4..6).each do |i|
+  blank = Marshal.load(Marshal.dump(form_structure))
+  test_cases << { id: i, date: Time.zone.today, form_json: blank }
+end
+
+# Create seeds
+
+test_cases.each do |tc|
+  # Saved claim stub
+  saved = SavedClaim::DisabilityCompensation.new(
+    user_account_id: user_account.id,
+    form_id: '21-526EZ',
+    form: { 'seed' => tc[:id] }.to_json
+  )
+  saved.save!(validate: false)
+
+  submission = Form526Submission.create!(
+    id: tc[:id],
+    user_uuid: user_account.id,
+    user_account_id: user_account.id,
+    saved_claim_id: saved.id,
+    auth_headers_json: { 'va_eauth_dodedipnid' => 'seed' }.to_json,
+    form_json: tc[:form_json].to_json,
+    created_at: tc[:date].to_time,
+    updated_at: tc[:date].to_time
+  )
+
+  # Log what's in each form for debugging
+  form0781_incidents = tc[:form_json]['form0781']['form0781']['incidents']&.size || 0
+  form0781v2_incidents = tc[:form_json]['form0781']['form0781v2']['incidents']&.size || 0
```

Comment:
this comment isn't accurate, right? It appears you're logging the number of incidents and not the contents of each form. 😳 The contents of a form would not be good to log.

Reply:
These are seeds for local dev testing, not production data.